GenderPayGap.WebUI/Views/ManageOrganisations/AllOrganisationReports.cshtml

```diff
@@ -0,0 +1,329 @@
+        {
+            Head = desktopHeader,
+            Rows = desktopDisplayRows,
+            Classes = "gpg-govuk-hideOnMobile govuk-grid-column-full"
+        })
+
+    @await Html.GovUkTable(new TableGovUkViewModel
+        {
+            Head = new List<TableCellViewModel>(),
+            Rows = mobileDisplayRows,
+            Classes = "gpg-govuk-hideOnDesktop govuk-grid-column-full"
+        })
+    
+    @{
+        var nextLink = new PaginationLinkViewModel()
+        {
+            Text = "Next page",
+            LabelText = $"{Model.CurrentPage + 1} of {Model.TotalPages}",
+            Href = Url.Action("AllOrganisationReportsGet", new {encryptedOrganisationId, page = Model.CurrentPage + 1})
+        };
+    
+        var previousLink = new PaginationLinkViewModel()
+        {
+            Text = "Previous page",
+            LabelText = $"{Model.CurrentPage - 1} of {Model.TotalPages}",
+            Href = Url.Action("AllOrganisationReportsGet", new {encryptedOrganisationId, page = Model.CurrentPage - 1})
+        };
+    
+        if (Model.TotalPages < 2 || Model.CurrentPage == Model.TotalPages)
+            nextLink = null;
+        if (Model.CurrentPage < 2)
+            previousLink = null;
+    }
+    @await Html.GovUkPagination(new PaginationViewModel()
+        {
+            Next = nextLink,
+            Previous = previousLink,
+        })
+    </div>
+</div>
+
+<script>
+    var parentForSelect = document.getElementById("filteringArea");
+    
+    var selectLabel = document.createElement("label");
+    selectLabel.className = "govuk-label";
+    selectLabel.for = "yearFilter";
+    selectLabel.textContent = "Filter for a snapshot date";
+    
+    var elements = document.getElementsByClassName("yearTextForFiltering");
+    
+    var selectList = document.createElement("select");
+    selectList.className = "govuk-select govuk-!-margin-bottom-3";
+    selectList.id = "filterYearSelect";
+    selectList.addEventListener("change",function (e){
+        for (var i = 0; i < elements.length; i++){
+            var tableRow = elements[i].closest("tr");
+            tableRow.hidden = true;
+            
+            if (elements[i].innerHTML.includes(e.target.value) || e.target.value === "Show All"){
+                tableRow.hidden = false;
+            }
+        }
+    }) 
+   
+    parentForSelect.appendChild(selectLabel);
+    parentForSelect.appendChild(selectList);
+
+    var showAll = document.createElement("option");
+    showAll.value = "Show All";
+    showAll.text = "Show All";
+    showAll.selected = true;
+    selectList.appendChild(showAll);
+    
+    var arrayOfYears = JSON.parse(document.getElementById("selectOptionsFromReports").value);
+    for (var i = 0; i < arrayOfYears.length; i++){
+         var option = document.createElement("option");
+         option.value = arrayOfYears[i];
+         option.text = arrayOfYears[i];
+         selectList.appendChild(option);
+    }
+</script>

```

Comment:
Is this exactly the same as on the other page with filtering? If so (or if it's so close we can make it the same) can we split it out into a partial?

Reply:
They are subtly different, in terms of label and the existence of a hint div on the employer page, 
it's debatable if worth doing for these two, I'm thinking if it comes up again in a future ticket then it's worth giving it a view model and a partial?
I am of course happy to do it if you feel it is worth doing at this point.

Reply:
Maybe a tech debt ticket instead then. Happy to leave for now as long as there's a reminder to improve it somewhere.